cmd/snap/cmd_debug_seeding.go

```diff
@@ -0,0 +1,136 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/snapcore/snapd/interfaces"
+)
+
+type cmdSeeding struct {
+	clientMixin
+}
+
+func init() {
+	cmd := addDebugCommand("seeding",
+		"(internal) obtain seeding and preseeding details",
+		"(internal) obtain seeding and preseeding details",
+		func() flags.Commander {
+			return &cmdSeeding{}
+		}, nil, nil)
+	cmd.hidden = true
+}
+
+func (x *cmdSeeding) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+	var resp struct {
+		Seeded               bool        `json:"seeded,omitempty"`
+		Preseeded            bool        `json:"preseeded,omitempty"`
+		PreseedStartTime     time.Time   `json:"preseed-start-time,omitempty"`
+		PreseedTime          time.Time   `json:"preseed-time,omitempty"`
+		SeedStartTime        time.Time   `json:"seed-start-time,omitempty"`
+		SeedRestartTime      time.Time   `json:"seed-restart-time,omitempty"`
+		SeedTime             time.Time   `json:"seed-time,omitempty"`
+		PreseedSystemKey     interface{} `json:"preseed-system-key,omitempty"`
+		SeedRestartSystemKey interface{} `json:"seed-restart-system-key,omitempty"`
+	}
+	if err := x.client.DebugGet("seeding", &resp, nil); err != nil {
+		return err
+	}
+
+	w := tabWriter()
+
+	// show seeded and preseeded keys
+	fmt.Fprintf(w, "seeded:\t%v\n", resp.Seeded)
+	fmt.Fprintf(w, "preseeded:\t%v\n", resp.Preseeded)
+
+	// calculate the time spent preseeding (if preseeded) and seeding
+	// for the preseeded case, we use the seed-restart-time as the start time
+	// to show how long we spent only after booting the preseeded image
+	var seedDuration time.Duration
+	if resp.Preseeded {
+		preseedDuration := resp.PreseedTime.Sub(resp.PreseedStartTime)
+		fmt.Fprintf(w, "image-preseeding:\t%v\n", preseedDuration)
```

Comment:
Maybe we could `.Round(time.Millisecond)` here and when printing seed completion.